cdc/sink/mysql.go

```diff
@@ -1101,3 +1104,160 @@ func buildColumnList(names []string) string {
+		credential := security.Credential{
+			CAPath:   sinkURI.Query().Get("ssl-ca"),
+			CertPath: sinkURI.Query().Get("ssl-cert"),
+			KeyPath:  sinkURI.Query().Get("ssl-key"),
+		}
+		tlsCfg, err := credential.ToTLSConfig()
+		if err != nil {
+			return nil, errors.Annotate(err, "fail to open MySQL connection")
+		}
+		name := "cdc_mysql_tls" + "syncpoint" + id
+		err = dmysql.RegisterTLSConfig(name, tlsCfg)
+		if err != nil {
+			return nil, errors.Annotate(err, "fail to open MySQL connection")
+		}
+		tlsParam = "?tls=" + name
+	}
+
+	// dsn format of the driver:
+	// [username[:password]@][protocol[(address)]]/dbname[?param1=value1&...&paramN=valueN]
+	username := sinkURI.User.Username()
+	password, _ := sinkURI.User.Password()
+	port := sinkURI.Port()
+	if username == "" {
+		username = "root"
+	}
+	if port == "" {
+		port = "4000"
+	}
+
+	dsnStr := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", username, password, sinkURI.Hostname(), port, tlsParam)
+	dsn, err := dmysql.ParseDSN(dsnStr)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	tz := util.TimezoneFromCtx(ctx)
+	// create test db used for parameter detection
+	if dsn.Params == nil {
+		dsn.Params = make(map[string]string, 1)
+	}
+	dsn.Params["time_zone"] = fmt.Sprintf(`"%s"`, tz.String())
+	testDB, err := sql.Open("mysql", dsn.FormatDSN())
+	if err != nil {
+		return nil, errors.Annotate(
+			cerror.WrapError(cerror.ErrMySQLConnectionError, err), "fail to open MySQL connection when configuring sink")
+	}
+	defer testDB.Close()
+	dsnStr, err = configureSinkURI(ctx, dsn, tz, params, testDB)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	syncDB, err = sql.Open("mysql", dsnStr)
+	if err != nil {
+		return nil, errors.Annotate(err, "Open database connection failed")
+	}
+	err = syncDB.PingContext(ctx)
+	if err != nil {
+		return nil, errors.Annotate(err, "fail to open MySQL connection")
+	}
+
+	log.Info("Start mysql syncpoint sink")
+	syncpointSink := &mysqlSyncpointSink{
+		db: syncDB,
+	}
+
+	return syncpointSink, nil
+}
+
+func (s *mysqlSyncpointSink) CreateSynctable(ctx context.Context) error {
+	database := "TiCDC"
+	tx, err := s.db.BeginTx(ctx, nil)
+	if err != nil {
+		log.Error("create sync table: begin Tx fail", zap.Error(err))
+		return cerror.WrapError(cerror.ErrMySQLTxnError, err)
+	}
+	_, err = tx.Exec("CREATE DATABASE IF NOT EXISTS " + database)
+	if err != nil {
+		err2 := tx.Rollback()
+		if err2 != nil {
+			log.Error(cerror.WrapError(cerror.ErrMySQLTxnError, err2).Error())
+		}
+		return cerror.WrapError(cerror.ErrMySQLTxnError, err)
+	}
+	_, err = tx.Exec("USE " + database)
+	if err != nil {
+		err2 := tx.Rollback()
+		if err2 != nil {
+			log.Error(cerror.WrapError(cerror.ErrMySQLTxnError, err2).Error())
+		}
+		return cerror.WrapError(cerror.ErrMySQLTxnError, err)
+	}
+	_, err = tx.Exec("CREATE TABLE  IF NOT EXISTS syncpoint (cf varchar(255),primary_ts varchar(18),secondary_ts varchar(18),PRIMARY KEY ( `cf`, `primary_ts` ) )")
```

Comment:
also make the table name `syncpoint` a constant, and add version to it, such as `syncpoint_v1`

Reply:
You means make `syncpoint_v1` as a constant and use this constant as our table name in the DML?

Reply:
yes